source/common/network/connection_impl.h

```diff
@@ -114,10 +115,11 @@ class ConnectionImpl : public virtual Connection,
 private:
   // clang-format off
   struct InternalState {
-    static const uint32_t ReadEnabled              = 0x1;
-    static const uint32_t Connecting               = 0x2;
-    static const uint32_t CloseWithFlush           = 0x4;
-    static const uint32_t ImmediateConnectionError = 0x8;
+    static const uint32_t ReadEnabled              = 0x01;
+    static const uint32_t Connecting               = 0x02;
+    static const uint32_t CloseWithFlush           = 0x04;
+    static const uint32_t ImmediateConnectionError = 0x08;
+    static const uint32_t Unconnected              = 0x10;
```

Comment:
What if we just start client connections in the Connecting state? Is there any downside to that? Do we need Unconnected state for anything?